app/controllers/sites_controller.rb

```diff
@@ -38,7 +39,9 @@ def show
 
     host_list = hosts.split(',')
     host_list.each do |host|
-      Host.create(hostname: host, cname: host, site: @site).save
+      Host.find_or_create_by(hostname: host.strip, site: @site) do |h|
+        h.cname = host.strip
+      end
```

Comment:
Could we maybe have a test to cover this?

Reply:
yeah, good point, I can extend one of the existing tests I think.

Reply:
Done, turns out the test for host name adding was broken anyway! So I've fixed it and added it into the update test as well.